### PR TITLE
Percent margins

### PR DESCRIPTION
**Pull request by vandyshev**
Add percent support for margins

### CODE REVIEW

**Comment by lucdion** [comment]
Thanks @vandyshev for your PR 🎉, I'll check your PR, but don't worry about the failure of travis, I have fixed it in my latest PR

File: Sources/Impl/PinLayoutImpl.swift [hunk 1]
```diff
@@ -818,47 +818,99 @@ class PinLayoutImpl: PinLayout {
...
+    }
     
     @discardableResult
     func marginEnd(_ value: CGFloat) -> PinLayout {
         return isLTR() ? marginRight(value) : marginLeft(value)
     }
 
+    @discardableResult
+    func marginEnd(_ percent: Percent) -> PinLayout {
+        return isLTR() ? marginRight(percent) : marginLeft(percent)
```
Comment (lucdion) [inline Sources/Impl/PinLayoutImpl.swift]: Here if there is a warning, the message will display the wrong context (either marginRight or marginLeft), it should display "marginEnd".

File: Sources/Impl/PinLayoutImpl.swift [hunk 2]
```diff
@@ -818,47 +818,99 @@ class PinLayoutImpl: PinLayout {
...
     @discardableResult
     func marginHorizontal(_ value: CGFloat) -> PinLayout {
         marginLeft = value
         marginRight = value
         return self
     }
 
+    @discardableResult
+    func marginHorizontal(_ percent: Percent) -> PinLayout {
+        return marginLeft(percent).marginRight(percent)
```
Comment (lucdion) [inline Sources/Impl/PinLayoutImpl.swift]: Wrong context here also

File: Sources/Impl/PinLayoutImpl.swift [hunk 3]
```diff
@@ -818,47 +818,99 @@ class PinLayoutImpl: PinLayout {
...
     @discardableResult
     func marginVertical(_ value: CGFloat) -> PinLayout {
         marginTop = value
         marginBottom = value
         return self
     }
+
+    @discardableResult
+    func marginVertical(_ percent: Percent) -> PinLayout {
+        return marginTop(percent).marginBottom(percent)
```
Comment (lucdion) [inline Sources/Impl/PinLayoutImpl.swift]: here too, and few other below.

File: Tests/MarginsSpec.swift [hunk 4]
```diff
@@ -57,6 +57,13 @@ class MarginsSpec: QuickSpec {
                 expect(aView.frame).to(equal(CGRect(x: 140.0, y: 100.0, width: 100.0, height: 120.0)))
             }
         }
+
+        describe("the result of left&right percent margins when only the width is specified") {
```
Comment (lucdion) [inline Tests/MarginsSpec.swift]: Excellent unit tests, could you just add fews in the WarningSpec.

**Comment by vandyshev** [comment]
@lucdion Ok, will be fixed

**Pull request merged**